cetmix_tower_server/models/cx_tower_reference_mixin.py

```diff
@@ -66,7 +66,15 @@ def _generate_or_fix_reference(self, reference_source):
         # Check if the same reference already exists and add a suffix if yes
         counter = 1
         final_reference = reference
-        while self.search_count([("reference", "=", final_reference)]) > 0:
+        while (
```

Comment:
This method can be called when no record exists yet. Eg [here](https://github.com/cetmix/cetmix-tower/blob/50c240f9c4b628043fddaf76ee5e4d84c30866a1/cetmix_tower_server/models/cx_tower_reference_mixin.py#L122)

So we need to check id only in case if it's called for record. Something like:
```python
domain = [("reference", "=", final_reference)]
if self:
    domain.append(("id", "not in", self.ids)),
```

We are using `self.ids` here bacause this method can be also called for a recordset. Eg in the `write` method [here](https://github.com/cetmix/cetmix-tower/blob/50c240f9c4b628043fddaf76ee5e4d84c30866a1/cetmix_tower_server/models/cx_tower_reference_mixin.py#L157)

Reply:
Hey @norlinhenrik could you please check #108 ?
It features a fix for this issue.